### PR TITLE
improve socket to send message only for recipient

### PR DESCRIPTION
**Pull request by jaytranuts247**
- Improve socket to send only for recipient

### CODE REVIEW

File: client/src/store/utils/setSocketAuthToken.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import socket from "../../socket";
+
+export const setSocketAuthToken = () => {
+  if (!socket.auth || !socket.auth.token) {
+    console.log("set token to socket ");
```
Comment (sonipranjal) [inline client/src/store/utils/setSocketAuthToken.js:5]:
```markdown
While `console.logs` are great for testing, they are best left out of code that is being pushed to production 🙂
```

**Review by sonipranjal — COMMENTED** [review]
One more thing we'd like to see is how we check for onlineUsers. 
This is currently an O(n) operation, which isn't the most efficient. 
Is there another way we can structure this to improve on efficiency?

File: server/bin/www [hunk 2]
```diff
@@ -31,37 +32,78 @@ const server = http.createServer(app);
...
-      onlineUsers.push(id);
+  console.log("socket server connected");
+
+  io.emit("socketId-collect");
+
+  socket.on("update-socketId", ({ userId, socketId }) => {
+    if (userId && socketId)
+      onlineUsers.set(userId, {
+        userId,
+        socketId,
```
Comment (dericf) [inline server/bin/www:61]: Nitpick: for onlineUsers, you don't need to save the userId in the object, since you already have it as the key.

**Review by dericf — APPROVED** [review]
Nice 👍 


You do **not** need to make this change, but one comment I have is about the use of `update-socketId` and `socketId-collect` events 🤔  It seems that is adding unnecessary complexity to the socket logic. A cleaner way to handle this imo would be to map userId to an array of sockets instead of a single socket - that way if they disconnect, or log in via multiple devices/browsers, each user would have a list of sockets instead of just one.

In any case, this PR is approved 🙂

**Pull request merged**